lwgenerate/azure/azure.go

```diff
@@ -0,0 +1,250 @@
+	blocks := []*hclwrite.Block{}
+	//attrs := map[string]interface{}{}
+	provider, err := lwgenerate.NewProvider(
+		"azuread",
+	).ToBlock()
+
+	if err != nil {
+		return nil, err
+	}
+
+	blocks = append(blocks, provider)
+	return blocks, nil
+}
+
+func createAzureRMProvider() ([]*hclwrite.Block, error) {
+	blocks := []*hclwrite.Block{}
+	attrs := map[string]interface{}{}
+
+	provider, err := lwgenerate.NewProvider(
+		"azurerm",
+	).ToBlock()
+
+	if err != nil {
+		return nil, err
+	}
+	// Create the features block
+	featuresBlock, err := lwgenerate.HclCreateGenericBlock("features", []string{}, attrs)
+	provider.Body().AppendBlock(featuresBlock)
+
+	if err != nil {
+		return nil, err
+	}
+
+	blocks = append(blocks, provider)
+	return blocks, nil
+}
+
+func createLaceworkAZADModule() ([]*hclwrite.Block, error) {
+	blocks := []*hclwrite.Block{}
+
+	provider, err := lwgenerate.NewModule(
+		"az_ad_application",
+		lwgenerate.LWAzureADSource,
+		lwgenerate.HclModuleWithVersion(lwgenerate.LWAzureADVersion),
+	).ToBlock()
+
+	if err != nil {
+		return nil, err
+	}
+
+	blocks = append(blocks, provider)
+	return blocks, nil
+}
+
+func createConfig(args *GenerateAzureTfConfigurationArgs) ([]*hclwrite.Block, error) {
+	blocks := []*hclwrite.Block{}
+	if args.Config {
+		attributes := map[string]interface{}{}
+		moduleDetails := []lwgenerate.HclModuleModifier{}
+
+		if args.ConfigIntegrationName != "" {
+			attributes["lacework_integration_name"] = args.ConfigIntegrationName
+		}
+
+		// Set default values
+		attributes["use_existing_ad_application"] = true
+		attributes["application_id"] = "module.az_ad_application.application_id"
+		attributes["application_password"] = "module.az_ad_application.application_password"
+		attributes["service_principal_id"] = "module.az_ad_application.service_principal_id"
+
+		moduleDetails = append(moduleDetails,
+			lwgenerate.HclModuleWithAttributes(attributes),
+		)
+
+		moduleBlock, err := lwgenerate.NewModule(
+			"az_config",
+			lwgenerate.LWAzureConfigSource,
+			append(moduleDetails, lwgenerate.HclModuleWithVersion(lwgenerate.LWAzureConfigVersion))...,
+		).ToBlock()
+
+		if err != nil {
+			return nil, err
+		}
+		blocks = append(blocks, moduleBlock)
+	}
+
+	return blocks, nil
+}
+
+func createActivityLog(args *GenerateAzureTfConfigurationArgs) ([]*hclwrite.Block, error) {
+	blocks := []*hclwrite.Block{}
+	if args.ActivityLog {
+		attributes := map[string]interface{}{}
+		moduleDetails := []lwgenerate.HclModuleModifier{}
+
+		if args.ActivityLogIntegrationName != "" {
+			attributes["lacework_integration_name"] = args.ActivityLogIntegrationName
+		}
+
+		// Set default values
+		attributes["use_existing_ad_application"] = true
+		attributes["application_id"] = "module.az_ad_application.application_id"
+		attributes["application_password"] = "module.az_ad_application.application_password"
+		attributes["service_principal_id"] = "module.az_ad_application.service_principal_id"
```

Comment:
Are these defaults correct? https://github.com/lacework/terraform-azure-activity-log/tree/main/examples/default-activity-log

Reply:
Again this is in the example, we create a an Active Directory application and reference that, I have added a flag for the user to use an existing AD. The terraform code will be generated accordingly depending on user choice